lib/Onyx.ts

```diff
@@ -100,15 +98,15 @@ function connect<TKey extends OnyxKey>(options: ConnectOptions<TKey>): number {
             // If the mapping is connected to an onyx key that is not a collection
             // we can skip the call to getAllKeys() and return an array with a single item
             if (Boolean(mapping.key) && typeof mapping.key === 'string' && !mapping.key.endsWith('_') && cache.storageKeys.has(mapping.key)) {
-                return [mapping.key];
+                return new Set([mapping.key]);
```

Comment:
Why this change?

Reply:
If you look closely at the logic of connect function, and precisely at this then() block:
```ts
.then(() => {
  // Performance improvement
  // If the mapping is connected to an onyx key that is not a collection
  // we can skip the call to getAllKeys() and return an array with a single item
  if (Boolean(mapping.key) && typeof mapping.key === 'string' && !mapping.key.endsWith('_') && cache.storageKeys.has(mapping.key)) {
      return new Set([mapping.key]);
  }
  return OnyxUtils.getAllKeys();
})
```

Before it was returning either an `Array` or a `Set`, so now I consolidated the return type to always be a `Set`, this way next then block is easier to work with 👍